lcm/lcm_file.c

```diff
@@ -355,22 +355,20 @@ lcm_logprov_publish (lcm_logprov_t *lcm, const char *channel, const void *data,
     return 0;
 }
 
-static lcm_provider_vtable_t logprov_vtable;
+static lcm_provider_vtable_t logprov_vtable = {
+    .create      = lcm_logprov_create,
+    .destroy     = lcm_logprov_destroy,
+    .subscribe   = NULL,
+    .unsubscribe = NULL,
+    .publish     = lcm_logprov_publish,
+    .handle      = lcm_logprov_handle,
+    .get_fileno  = lcm_logprov_get_fileno
+};
 static lcm_provider_info_t logprov_info;
 
-
 void
 lcm_logprov_provider_init (GPtrArray * providers)
 {
-// Microsoft VS compiler issues. Can't do this statically
```

Comment:
That would be safe; the "issue" may still exist on Windows, but screw em, it's Windows' fault for having a lousy compiler. It would be nice to know if it's still a problem on Windows, though, or if it's safe to take the patch as-is.